packages/angular_devkit/build_optimizer/src/transforms/wrap-enums.ts

```diff
@@ -112,11 +112,16 @@ function visitBlockStatements(
             oIndex++;
           }
 
-          const enumStatements = findStatements(name, statements, oIndex + 1);
+          const enumStatements = findStatements(name, statements, oIndex);
           if (!enumStatements) {
             continue;
           }
 
+          // first item is typically the const, var or export which is not needed
+          // as it will be created later on
+          // export var ChangeDetectionStrategy = {};
+          enumStatements.shift();
```

Comment:
Why the shift instead of the plus one above?

Reply:
Because in `findStatements` there is already a `+1` https://github.com/angular/angular-cli/blob/66df0e77004d626b8aef5ade792373902b640b73/packages/angular_devkit/build_optimizer/src/transforms/wrap-enums.ts#L327 and adding a +1 to the above would skip a statement. example


Having these statements
```
VariableStatement <-- current
IfStatement <-- findStatements will start iterating from this. 
ExpressionStatement <-- (if we add +1 to oIndex) it will skip to this statement, which cause incorrect encapsulations as we are not validating that the above statement is valid and should be wrapped.
ExpressionStatement
```

Reply:
In that case, why do we need the shift?  It should already be past the variable statement.

Reply:
Cause `findStatements` will iterate over the next statements to validate them. But will also return the statement at `oIndex`

Maybe can do something like `findStatements(name, statements, oIndex, offset);`?

Reply:
That'll work.
This is the change that actually fixes the referenced issue, correct?
With the visit change will this actually be tested by the newly added test?

Reply:
> This is the change that actually fixes the referenced issue, correct?

Yes
> With the visit change will this actually be tested by the newly added test?

Added test